scripts/ci/check_compliance.py

```diff
@@ -658,6 +659,32 @@ def check_no_undef_within_kconfig(self, kconf):
         if undef_ref_warnings:
             self.failure(f"Undefined Kconfig symbols:\n\n {undef_ref_warnings}")
 
+    def check_soc_name_sync(self, kconf):
+        root_args = argparse.Namespace(**{'soc_roots': [Path(ZEPHYR_BASE)]})
+        v2_systems = list_hardware.find_v2_systems(root_args)
+
+        soc_names = {soc.name for soc in v2_systems.get_socs()}
+
+        soc_kconfig_names = set()
+        for node in kconf.node_iter():
+            if isinstance(node.item, kconfiglib.Symbol) and node.item.name == "SOC":
+                n = node.item
+                for d in n.defaults:
+                    soc_kconfig_names.add(d[0].name)
+
+        soc_name_warnings = []
+        for name in soc_names:
+            if name not in soc_kconfig_names:
+                soc_name_warnings.append(f"soc name: {name} not found in CONFIG_SOC defaults.")
```

Comment:
This loops through `soc_names - soc_kconfig_names`, but I think `soc_kconfig_names - soc_names` should be checked as well, to ensure that the two sets are equal.

Reply:
Gotcha. I just thought the main purpose was to catch things like #69308, for which you'd have to test the opposite.

Reply:
I actually caught those with current implementation 😉

Reply:
Fair enough :) In this case, it was possible to figure out the problem from either direction.